### PR TITLE
Custom tsconfig.json for angular apps.

### PR DESCRIPTION
**Pull request by igor-dv**
Issue: one of #2645 

## What I did

Added ability to create custom `tsconfig.json` in the storybook config dir.

## How to test
Add custom `tsconfig.json` to the .storybook dir (I've added already one in angular-cli example)

### CODE REVIEW

**Comment by amcdnl** [comment]
@igor-dv - Whats the API for this?

**Comment by igor-dv** [comment]
@amcdnl just put `tsconfig.json` to the `.storybook` dir (like custom babelrc or webpack.config.js)

**Comment by amcdnl** [comment]
@igor-dv - Can you add this in the docs here? https://storybook.js.org/basics/guide-angular/

**Review by amcdnl — APPROVED** [review]
LGTM

File: app/angular/src/server/config.js [hunk 1]
```diff
@@ -15,6 +16,9 @@ export default function(configType, baseConfig, configDir) {
   const babelConfig = loadBabelConfig(configDir);
   config.module.rules[0].query = babelConfig;
 
+  const tsOptions = loadTsConfig(configDir);
+  config.module.rules[1].loaders[0].options = tsOptions;
```
Comment (igor-dv) [inline app/angular/src/server/config.js]: Sorry if it looks ugly, IMO need some work moving webpack configurations to the core.

**Comment by igor-dv** [comment]
> Can you add this in the docs here? https://storybook.js.org/basics/guide-angular/

Will add in a separate PR then (or maybe someone will add =)) )

**Comment by eagor** [comment]
> > Can you add this in the docs here? https://storybook.js.org/basics/guide-angular/
> 
> Will add in a separate PR then (or maybe someone will add =)) )

is it working only for Angular? or React as well?

**Pull request merged**